### PR TITLE
refactor: add tests and refactor aggregators

### PR DESCRIPTION
**Pull request by Judge40**
Add unit tests and refactor the doctor recommendation aggregation
strategies.

NO-TICKET

### CODE REVIEW

File: src/test/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/DoctorRecommendationAggregationStrategyTest.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  @BeforeEach
+  void setUp() {
+    objectMapper = new ObjectMapper();
+    objectMapper.registerModules(new JavaTimeModule());
+    aggregationStrategy = new DoctorRecommendationAggregationStrategy(objectMapper);
+  }
+
+  @Test
+  void shouldAggregateRecommendationWhenCoreFound() {
+    var camelContext = new DefaultCamelContext();
```
[the remark this answers is not in the document]
Comment (Judge40) [reply src/test/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/DoctorRecommendationAggregationStrategyTest.java:74]: So you're saying reval should be refactored to match everywhere else? :trollface:

File: src/test/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/DoctorRecommendationAggregationStrategyTest.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    final var traineeRecommendation = new TraineeRecommendationDto();
+    traineeRecommendation.setCctDate(OLD_CCT_DATE);
+    traineeRecommendation.setCurrentGrade(OLD_GRADE);
+    traineeRecommendation.setDesignatedBody(DESIGNATED_BODY);
+    traineeRecommendation.setFullName(FULL_NAME);
+    traineeRecommendation.setGmcNumber(GMC_NUMBER);
+    traineeRecommendation.setGmcSubmissionDate(GMC_SUBMISSION_DATE);
+    traineeRecommendation.setProgrammeMembershipType(OLD_PROGRAMME_MEMBERSHIP_TYPE);
+    traineeRecommendation.setUnderNotice(UNDER_NOTICE);
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/DoctorRecommendationAggregationStrategyTest.java:135]: If I'm being fussy, this block is common between tests.

File: src/test/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/DoctorRecommendationSummaryAggregationStrategyTest.java [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    traineeInfo1.setGmcReferenceNumber(GMC_REF_NUMBER_1);
+    traineeInfo1.setLastUpdatedDate(LAST_UPDATED_DATE_1);
+    traineeInfo1.setSubmissionDate(SUBMISSION_DATE_1);
+
+    final var traineeSummary = new TraineeSummaryDto();
+    traineeSummary.setCountTotal(1);
+    traineeSummary.setCountUnderNotice(2);
+    traineeSummary.setTotalPages(3);
+    traineeSummary.setTotalResults(4);
+    traineeSummary.setTraineeInfo(List.of(traineeInfo1));
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/DoctorRecommendationSummaryAggregationStrategyTest.java:185]: Common code block.  My assumption would be to have a separate method for instantiating each class.

File: src/test/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/DoctorRecommendationSummaryAggregationStrategyTest.java [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    oldMessage.setBody(traineeSummary);
+    final var oldExchange = new DefaultExchange(camelContext);
+    oldExchange.setIn(oldMessage);
+
+    final var traineeCore = new TraineeCoreDto();
+    traineeCore.setCctDate(CCT_DATE);
+    traineeCore.setCurrentGrade(GRADE);
+    traineeCore.setGmcOutcome(GMC_OUTCOME);
+    traineeCore.setProgrammeMembershipType(PROGRAMME_MEMBERSHIP_TYPE);
+    traineeCore.setProgrammeName(PROGRAMME_NAME);
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/DoctorRecommendationSummaryAggregationStrategyTest.java:198]: Common code block.

File: src/test/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/DoctorRecommendationSummaryAggregationStrategyTest.java [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+    traineeInfo2.setGmcReferenceNumber(GMC_REF_NUMBER_2);
+    traineeInfo2.setLastUpdatedDate(LAST_UPDATED_DATE_2);
+    traineeInfo2.setSubmissionDate(SUBMISSION_DATE_2);
+
+    final var traineeSummary = new TraineeSummaryDto();
+    traineeSummary.setCountTotal(1);
+    traineeSummary.setCountUnderNotice(2);
+    traineeSummary.setTotalPages(3);
+    traineeSummary.setTotalResults(4);
+    traineeSummary.setTraineeInfo(List.of(traineeInfo1, traineeInfo2));
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/tis/revalidation/integration/router/aggregation/DoctorRecommendationSummaryAggregationStrategyTest.java:273]: De-duplicating here could use a parameterized method as in other services.

**Review by JosephKelly — APPROVED** [review]

**Pull request merged**